packages/react/src/modules/slack/components/SlackChannelCombobox/AddConnectedSlackChannelInput.tsx

```diff
@@ -0,0 +1,72 @@
+import { SlackChannelConnection } from "@knocklabs/client";
+import { useState } from "react";
+
+import { SlackIcon } from "../SlackIcon";
+
+import ConnectionErrorInfoBoxes from "./ConnectionErrorInfoBoxes";
+import "./styles.css";
+
+const AddConnectedSlackChannelInput = ({
+  isErrorState,
+  connectedChannels = [],
+  updateConnectedChannels,
+  connectedChannelsError,
+  connectedChannelsUpdating,
+}: {
+  isErrorState: boolean;
+  connectedChannels: SlackChannelConnection[];
+  updateConnectedChannels: (channels: SlackChannelConnection[]) => void;
+  connectedChannelsError: string | null;
+  connectedChannelsUpdating: boolean;
+}) => {
+  const [value, setValue] = useState<string | null>(null);
+  const [localError, setLocalError] = useState<string | null>(null);
+
+  const submitChannel = () => {
+    if (!value) {
+      return;
+    }
+
+    if (localError) {
+      setLocalError(null);
+    }
+
+    const alreadyConnected = connectedChannels.find(
+      (channel) => channel.channel_id === value,
+    );
+
+    if (alreadyConnected) {
+      setValue("");
+      return setLocalError(`"${value}" already connected.`);
+    }
+
+    const channelsToSendToKnock = [...connectedChannels, { channel_id: value }];
+    updateConnectedChannels(channelsToSendToKnock);
+    setValue("");
+  };
+
+  return (
+    <div style={{ display: "flex", gap: "4px", alignItems: "center" }}>
```

Comment:
Nit: Should clean these inline styles up (but can address it once the design is finalized)